toolkit/tools/pkg/imagecustomizerlib/createuuid.go

```diff
@@ -47,3 +51,61 @@ func convertUuidToString(uuid [UuidSize]byte) string {
 
 	return uuidStr
 }
+
+func extractImageUUID(imageConnection *ImageConnection) ([UuidSize]byte, string, error) {
+	var emptyUuid [UuidSize]byte
+
+	releasePath := filepath.Join(imageConnection.Chroot().RootDir(), "etc/image-customizer-release")
+	data, err := file.Read(releasePath)
+	if err != nil {
+		return emptyUuid, "", fmt.Errorf("failed to read %s:\n%w", releasePath, err)
+	}
+
+	lines := strings.Split(string(data), "\n")
+	var uuidStr string
+	for _, line := range lines {
+		if strings.HasPrefix(line, "IMAGE_UUID=") {
+			uuidStr = strings.Trim(strings.TrimPrefix(line, "IMAGE_UUID="), `"`)
+			break
+		}
+	}
+
+	if uuidStr == "" {
+		return emptyUuid, "", fmt.Errorf("IMAGE_UUID not found in %s", releasePath)
+	}
+
+	parsed, err := parseUuidString(uuidStr)
+	if err != nil {
+		return emptyUuid, "", fmt.Errorf("failed to parse IMAGE_UUID from string %s:\n%w", uuidStr, err)
+	}
+
+	return parsed, uuidStr, nil
+}
+
+func parseUuidString(s string) ([UuidSize]byte, error) {
+	var uuid [UuidSize]byte
+	n, err := fmt.Sscanf(s,
+		"%08x-%04x-%04x-%04x-%012x",
+		new(uint32),
```

Comment:
The results of `Sscanf` is stored in the pointer values passed to `Sscanf`. You should keep references to the pointers and then extract the bytes from those values.

Also, `Sscanf` will accept pointers to byte arrays.

Reply:
I guess in this case, it's better to remove the `Sscanf` entirely?